Cubical/Homotopy/BlakersMassey.agda

```diff
@@ -0,0 +1,671 @@
+              (fiber→←[q₀₀=q₁₀] q₁₀ r p) (fiber→←[q₁₁=q₁₀] q₁₀ r p)
+    fiber→←hypercube r p i j u v =
+      hcomp (λ l → λ { (i = i0) → fiber→←[q₀₀=q₁₀]-filler q₁₀ r p j u v l
+                     ; (i = i1) → fiber→←[q₁₁=q₁₀]-filler q₁₀ r p j u v l
+                     ; (j = i0) → fiber←[q₀₀=q₀₁=q₁₁]-filler q₁₀ r (fiber→[q₀₀=q₁₀=q₁₁] q₁₀ i r p .snd) i u v l
+                     ; (j = i1) → fiber→[q₀₀=q₁₀=q₁₁]-filler q₁₀ r p i l v u
+                     ; (u = i0) → push q₁₀ ((i ∨ (~ v ∧ l)) ∧ (~ v ∨ (i ∧ ~ l)))
+                     ; (u = i1) → fiber→[q₀₀=q₁₀=q₁₁] q₁₀ i r p .snd l v
+                     ; (v = i0) → push q₁₀ ((i ∨ l) ∧ ~ u)
+                     ; (v = i1) → push q₁₀ ((i ∧ ~ l) ∨ u) })
+            (push q₁₀ ((i ∨ (v ∧ u)) ∧ (v ∨ (i ∧ ~ u))))
+
+  {- Definitions of fiber←→ -}
+
+  module _
+    {x₀ : X}{y₁ : Y}(q₀₁ : Q x₀ y₁) where
+
+    {- (x₁ , q₁₁) = (x₀ , q₀₁) -}
+    module _
+      {y₀ : Y}(q₀₀ : Q x₀ y₀)
+      (r : inl x₀ ≡ inr y₁)
+      (p : push q₀₁ ≡ r) where
+
+      fiber←→[q₁₁=q₀₁]-filler : (i j k l : I) → Pushout
+      fiber←→[q₁₁=q₀₁]-filler i j k l' =
+        let p' = fiber←[q₁₁=q₀₁] q₀₁ q₀₀ r p .snd in
+        hfill (λ l → λ { (i = i0) → fiber→[q₀₀=q₁₀]-filler q₀₀ q₀₁ r p' j k l
+                       ; (i = i1) → fiber←[q₁₁=q₀₁]-filler q₀₁ q₀₀ r p l k j
+                       ; (j = i0) → push q₀₁ (k ∧ l)
+                       ; (j = i1) → p' l k
+                       ; (k = i0) → push q₀₀ (j ∧ ~ l)
+                       ; (k = i1) → push q₀₁ l })
+              (inS (push q₀₀ (j ∧ ~ k))) l'
+
+      fiber←→[q₁₁=q₀₁] : fiber→[q₀₀=q₁₀] q₀₀ q₀₁ r (fiber←[q₁₁=q₀₁] q₀₁ q₀₀ r p .snd) .snd ≡ p
+      fiber←→[q₁₁=q₀₁] i j k = fiber←→[q₁₁=q₀₁]-filler i j k i1
+
+    {- (y₀ , q₀₀) = (y₁ , q₀₁) -}
+    module _
+      {x₁ : X}(q₁₁ : Q x₁ y₁)
+      (r : inl x₀ ≡ inr y₁)
+      (p : push q₀₁ ≡ r) where
+
+      fiber←→[q₀₀=q₀₁]-filler : (i j k l : I) → Pushout
+      fiber←→[q₀₀=q₀₁]-filler i j k l' =
+        let p' = fiber←[q₀₀=q₀₁] q₀₁ q₁₁ r p .snd in
+        hfill (λ l → λ { (i = i0) → fiber→[q₁₁=q₁₀]-filler q₁₁ q₀₁ r p' j k l
+                       ; (i = i1) → fiber←[q₀₀=q₀₁]-filler q₀₁ q₁₁ r p l k j
+                       ; (j = i0) → push q₀₁ (k ∨ ~ l)
+                       ; (j = i1) → p' l k
+                       ; (k = i0) → push q₀₁ (~ l)
+                       ; (k = i1) → push q₁₁ (~ j ∨ l) })
+              (inS (push q₁₁ (~ j ∨ ~ k))) l'
+
+      fiber←→[q₀₀=q₀₁] : fiber→[q₁₁=q₁₀] q₁₁ q₀₁ r (fiber←[q₀₀=q₀₁] q₀₁ q₁₁ r p .snd) .snd ≡ p
+      fiber←→[q₀₀=q₀₁] i j k = fiber←→[q₀₀=q₀₁]-filler i j k i1
+
+    {- q₀₀ = q₀₁ = q₁₁ -}
+    fiber←→hypercube :
+        (r : inl x₀ ≡ inr y₁)
+      → (p : push q₀₁ ≡ r)
+      → PathP (λ i → fiber→[q₀₀=q₁₀=q₁₁] q₀₁ i r (fiber←[q₀₀=q₀₁=q₁₁] q₀₁ i r p .snd) .snd ≡ p)
+              (fiber←→[q₁₁=q₀₁] q₀₁ r p) (fiber←→[q₀₀=q₀₁] q₀₁ r p)
+    fiber←→hypercube r p i j u v =
```

Comment:
Wow, this is pretty wild!

Reply:
haha not my fault. I was forced by the laws of nature.